tools/osv-scanner/cmd/scanner.go

```diff
@@ -119,18 +157,80 @@ func printResults(query osv.BatchedQuery, resp *osv.BatchedResponse) {
 }
 
 // TODO(ochang): Machine readable output format.
-// TODO(ochang): Ability to specify type of input.
 func main() {
-	flag.Usage = usage
-	flag.Parse()
-
 	var query osv.BatchedQuery
 
-	for _, arg := range flag.Args() {
-		if err := scan(&query, arg); err != nil {
-			log.Printf("scan failed: %v\n", err)
-			return
-		}
+	app := &cli.App{
+		Name:  "osv-scanner",
+		Usage: "scans various mediums for dependencies and matches it against the OSV database",
+		Flags: []cli.Flag{
+			&cli.StringSliceFlag{
+				Name:      "docker",
+				Aliases:   []string{"D"},
+				Usage:     "scan docker image with this name",
+				TakesFile: false,
+			},
+			&cli.StringSliceFlag{
+				Name:      "lockfile",
+				Aliases:   []string{"L"},
+				Usage:     "scan package lockfile on this path",
+				TakesFile: true,
+			},
+			&cli.StringSliceFlag{
+				Name:      "sbom",
+				Aliases:   []string{"S"},
+				Usage:     "scan sbom file on this path",
+				TakesFile: true,
+			},
+			&cli.StringSliceFlag{
+				Name:      "git",
+				Aliases:   []string{"G"},
+				Usage:     "scan for git repository in this directory",
+				TakesFile: true,
+			},
+		},
+		ArgsUsage: "[directory1 directory2...]",
+		Action: func(context *cli.Context) error {
+			containers := context.StringSlice("docker")
+			for _, container := range containers {
+				// TODO: Automatically figure out what docker base image
+				// and scan appropriately.
+				scanDebianDocker(&query, container)
+			}
+
+			lockfiles := context.StringSlice("lockfile")
+			for _, lockfile := range lockfiles {
+				err := scanLockfile(&query, lockfile)
+				if err != nil {
+					return err
+				}
+			}
+
+			sboms := context.StringSlice("sbom")
+			for _, sbom := range sboms {
+				err := scanSbomFile(&query, sbom)
```

Comment:
nit: in Go we typically name acronyms like SBOMs in all caps.

See https://github.com/golang/go/wiki/CodeReviewComments#initialisms